src/ast/nodes/ForStatement.ts

```diff
@@ -25,13 +25,7 @@ export default class ForStatement extends Statement {
 		if (this.init) this.init.initialise(this.scope);
 		if (this.test) this.test.initialise(this.scope);
 		if (this.update) this.update.initialise(this.scope);
-
-		if (this.body.type === 'BlockStatement') {
-			this.body.initialiseScope(this.scope);
-			this.body.initialiseChildren(this.scope);
-		} else {
-			this.body.initialise(this.scope);
-		}
+		this.body.initialise(this.scope);
```

Comment:
Are you sure this simplifies?

Reply:
As a matter of fact it does. `initialise` just calls in order `initialiseScope`, `initialiseNode` and `initialiseChildren`. And `initialiseNode` is empty for BlockStatements.